console/src/telemetry/Actions.ts

```diff
@@ -38,46 +47,46 @@ const setConsoleOptsInDB = (opts, successCb, errorCb) => (
     ),
   };
 
-  return dispatch(requestAction(url, options)).then(
-    data => {
+  return (dispatch(requestAction(url, options)) as any).then(
+    (data: object) => {
       if (successCb) {
         successCb(data);
       }
     },
-    error => {
+    (error: Error) => {
       if (errorCb) {
         errorCb(error);
       }
     }
   );
 };
 
-const telemetryNotificationShown = () => dispatch => {
+const telemetryNotificationShown = () => (dispatch: (arg: Partial<SetNotificationShowAction>) => void) => {
```

Comment:
```suggestion
const telemetryNotificationShown = () => (dispatch: Dispatch<TelemetryActionTypes>) => void) => {
```
And you can update `SetNotificationShowAction` to:
```ts
interface SetNotificationShowAction {
  type: typeof SET_NOTIFICATION_SHOWN;
  data?: object;
}
```